plz check this massage on compile.

For greenplum filedump work

GPDB :  version 6.17.3 build commit:d2912fee614b350ea5781581b23a6697150935cc


make -f Makefile.gpdb download => Ok
make -f Makefile.gpdb => Fail

[gpadmin@mdw greenplum-db]$ source greenplum_path.sh
[gpadmin@mdw filedump-main]$ make -f Makefile.gpdb download
ENABLE_ZLIB: n
ENABLE_ZSTD: n
ENABLE_QUICKLZ: n
rm -rf gpdb_6_9.4
make -f Makefile.gpdb gpdb_6_9.4/timestamp.6.10.1
ENABLE_ZLIB: n
ENABLE_ZSTD: n
ENABLE_QUICKLZ: n
make[1]: Entering directory `/home/gpadmin/filedump-main'
rm -rf gpdb_6_9.4
mkdir -p gpdb_6_9.4/src/backend/utils/adt/ gpdb_6_9.4/src/backend/access/common/ gpdb_6_9.4/src/backend/cdb/
rm -f gpdb_6_9.4/src/backend/utils/adt/numutils.c gpdb_6_9.4/src/backend/access/common/heaptuple.c gpdb_6_9.4/src/backend/cdb/cdbappendonlystorageformat.c
curl --silent --output gpdb_6_9.4/src/backend/utils/adt/numutils.c https://raw.githubusercontent.com/greenplum-db/gpdb/6.10.1/src/backend/utils/adt/numutils.c; curl --silent --output gpdb_6_9.4/src/backend/access/common/heaptuple.c https://raw.githubusercontent.com/greenplum-db/gpdb/6.10.1/src/backend/access/common/heaptuple.c; curl --silent --output gpdb_6_9.4/src/backend/cdb/cdbappendonlystorageformat.c https://raw.githubusercontent.com/greenplum-db/gpdb/6.10.1/src/backend/cdb/cdbappendonlystorageformat.c;
touch gpdb_6_9.4/timestamp.6.10.1
make[1]: Leaving directory `/home/gpadmin/filedump-main'
[gpadmin@mdw filedump-main]$ make -f Makefile.gpdb
ENABLE_ZLIB: n
ENABLE_ZSTD: n
ENABLE_QUICKLZ: n
gcc -m64 -Wall -Wmissing-prototypes -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fno-aggressive-loop-optimizations -Wno-unused-but-set-variable -Wno-address -m64 -O3 -fargument-noalias-global -fno-omit-frame-pointer -g -std=gnu99   -Werror=uninitialized -Werror=implicit-function-declaration -I. -I./ -I/usr/local/greenplum-db-6.17.3/include/postgresql/server -I/usr/local/greenplum-db-6.17.3/include/postgresql/internal -D_GNU_SOURCE -I/usr/include/libxml2   -c -o decode.o decode.c
gcc -m64 -Wall -Wmissing-prototypes -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fno-aggressive-loop-optimizations -Wno-unused-but-set-variable -Wno-address -m64 -O3 -fargument-noalias-global -fno-omit-frame-pointer -g -std=gnu99   -Werror=uninitialized -Werror=implicit-function-declaration -I. -I./ -I/usr/local/greenplum-db-6.17.3/include/postgresql/server -I/usr/local/greenplum-db-6.17.3/include/postgresql/internal -D_GNU_SOURCE -I/usr/include/libxml2   -c -o pg_filedump.o pg_filedump.c
gcc -m64 -Wall -Wmissing-prototypes -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fno-aggressive-loop-optimizations -Wno-unused-but-set-variable -Wno-address -m64 -O3 -fargument-noalias-global -fno-omit-frame-pointer -g -std=gnu99   -Werror=uninitialized -Werror=implicit-function-declaration -I. -I./ -I/usr/local/greenplum-db-6.17.3/include/postgresql/server -I/usr/local/greenplum-db-6.17.3/include/postgresql/internal -D_GNU_SOURCE -I/usr/include/libxml2   -c -o stringinfo.o stringinfo.c
gcc -m64 -Wall -Wmissing-prototypes -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fno-aggressive-loop-optimizations -Wno-unused-but-set-variable -Wno-address -m64 -O3 -fargument-noalias-global -fno-omit-frame-pointer -g -std=gnu99   -Werror=uninitialized -Werror=implicit-function-declaration -I. -I./ -I/usr/local/greenplum-db-6.17.3/include/postgresql/server -I/usr/local/greenplum-db-6.17.3/include/postgresql/internal -D_GNU_SOURCE -I/usr/include/libxml2   -c -o gpdb_6_9.4/src/backend/utils/adt/numutils.o gpdb_6_9.4/src/backend/utils/adt/numutils.c
gcc -m64 -Wall -Wmissing-prototypes -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fno-aggressive-loop-optimizations -Wno-unused-but-set-variable -Wno-address -m64 -O3 -fargument-noalias-global -fno-omit-frame-pointer -g -std=gnu99   -Werror=uninitialized -Werror=implicit-function-declaration -I. -I./ -I/usr/local/greenplum-db-6.17.3/include/postgresql/server -I/usr/local/greenplum-db-6.17.3/include/postgresql/internal -D_GNU_SOURCE -I/usr/include/libxml2   -c -o gpdb_6_9.4/src/backend/access/common/heaptuple.o gpdb_6_9.4/src/backend/access/common/heaptuple.c
gcc -m64 -Wall -Wmissing-prototypes -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fno-aggressive-loop-optimizations -Wno-unused-but-set-variable -Wno-address -m64 -O3 -fargument-noalias-global -fno-omit-frame-pointer -g -std=gnu99   -Werror=uninitialized -Werror=implicit-function-declaration -I. -I./ -I/usr/local/greenplum-db-6.17.3/include/postgresql/server -I/usr/local/greenplum-db-6.17.3/include/postgresql/internal -D_GNU_SOURCE -I/usr/include/libxml2   -c -o gpdb_6_9.4/src/backend/cdb/cdbappendonlystorageformat.o gpdb_6_9.4/src/backend/cdb/cdbappendonlystorageformat.c
gcc -m64 -Wall -Wmissing-prototypes -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fno-aggressive-loop-optimizations -Wno-unused-but-set-variable -Wno-address -m64 -O3 -fargument-noalias-global -fno-omit-frame-pointer -g -std=gnu99   -Werror=uninitialized -Werror=implicit-function-declaration -I. -I./ -I/usr/local/greenplum-db-6.17.3/include/postgresql/server -I/usr/local/greenplum-db-6.17.3/include/postgresql/internal -D_GNU_SOURCE -I/usr/include/libxml2   -c -o gpdb.o gpdb.c
gpdb.c: In function ‘Dump_AppendOnlyStorage_ContentInfo’:
gpdb.c:224:2: error: unknown type name ‘Bytef’
  Bytef    *uncompressedDatum = NULL;
  ^
gpdb.c:326:13: warning: assignment from incompatible pointer type [enabled by default]
   headerPtr = uncompressedDatum;
             ^
make: *** [gpdb.o] Error 1

[gpadmin@mdw filedump-main]$ make -f Makefile.gpdb ENABLE_ZLIB=y
Makefile.gpdb:80: *** cannot find zlib.  Stop.
[gpadmin@mdw filedump-main]$ rpm -qa zlib
zlib-1.2.7-19.el7_9.x86_64

The same issue with gpdb 6.26.3.

```bash
[root@c4f3706b2f3a tmp]# ls -la /usr/local/greenplum-db/greenplum_path.sh
-rwxr-xr-x 1 root root 1252 Feb 15 14:12 /usr/local/greenplum-db/greenplum_path.sh
[root@c4f3706b2f3a tmp]# source /usr/local/greenplum-db/greenplum_path.sh
[root@c4f3706b2f3a tmp]# cd filedump-historical
[root@c4f3706b2f3a filedump-historical]# make -f Makefile.gpdb download
ENABLE_ZLIB: n
ENABLE_ZSTD: n
ENABLE_QUICKLZ: n
rm -rf gpdb_6_9.4
make -f Makefile.gpdb gpdb_6_9.4/timestamp.6.10.1
ENABLE_ZLIB: n
ENABLE_ZSTD: n
ENABLE_QUICKLZ: n
make[1]: Entering directory `/tmp/filedump-historical'
rm -rf gpdb_6_9.4
mkdir -p gpdb_6_9.4/src/backend/utils/adt/ gpdb_6_9.4/src/backend/access/common/ gpdb_6_9.4/src/backend/cdb/
rm -f gpdb_6_9.4/src/backend/utils/adt/numutils.c gpdb_6_9.4/src/backend/access/common/heaptuple.c gpdb_6_9.4/src/backend/cdb/cdbappendonlystorageformat.c
curl --silent --output gpdb_6_9.4/src/backend/utils/adt/numutils.c https://raw.githubusercontent.com/greenplum-db/gpdb/6.10.1/src/backend/utils/adt/numutils.c; curl --silent --output gpdb_6_9.4/src/backend/access/common/heaptuple.c https://raw.githubusercontent.com/greenplum-db/gpdb/6.10.1/src/backend/access/common/heaptuple.c; curl --silent --output gpdb_6_9.4/src/backend/cdb/cdbappendonlystorageformat.c https://raw.githubusercontent.com/greenplum-db/gpdb/6.10.1/src/backend/cdb/cdbappendonlystorageformat.c;
touch gpdb_6_9.4/timestamp.6.10.1
make[1]: Leaving directory `/tmp/filedump-historical'
[root@c4f3706b2f3a filedump-historical]# make -f Makefile.gpdb
ENABLE_ZLIB: n
ENABLE_ZSTD: n
ENABLE_QUICKLZ: n
gcc -m64 -Wall -Wmissing-prototypes -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fno-aggressive-loop-optimizations -Wno-unused-but-set-variable -Wno-address -m64 -O3 -fargument-noalias-global -fno-omit-frame-pointer -g -std=gnu99   -Werror=uninitialized -Werror=implicit-function-declaration -I. -I./ -I/usr/local/greenplum-db-6.26.3/include/postgresql/server -I/usr/local/greenplum-db-6.26.3/include/postgresql/internal -D_GNU_SOURCE -I/usr/include/libxml2   -c -o decode.o decode.c
gcc -m64 -Wall -Wmissing-prototypes -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fno-aggressive-loop-optimizations -Wno-unused-but-set-variable -Wno-address -m64 -O3 -fargument-noalias-global -fno-omit-frame-pointer -g -std=gnu99   -Werror=uninitialized -Werror=implicit-function-declaration -I. -I./ -I/usr/local/greenplum-db-6.26.3/include/postgresql/server -I/usr/local/greenplum-db-6.26.3/include/postgresql/internal -D_GNU_SOURCE -I/usr/include/libxml2   -c -o pg_filedump.o pg_filedump.c
gcc -m64 -Wall -Wmissing-prototypes -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fno-aggressive-loop-optimizations -Wno-unused-but-set-variable -Wno-address -m64 -O3 -fargument-noalias-global -fno-omit-frame-pointer -g -std=gnu99   -Werror=uninitialized -Werror=implicit-function-declaration -I. -I./ -I/usr/local/greenplum-db-6.26.3/include/postgresql/server -I/usr/local/greenplum-db-6.26.3/include/postgresql/internal -D_GNU_SOURCE -I/usr/include/libxml2   -c -o stringinfo.o stringinfo.c
gcc -m64 -Wall -Wmissing-prototypes -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fno-aggressive-loop-optimizations -Wno-unused-but-set-variable -Wno-address -m64 -O3 -fargument-noalias-global -fno-omit-frame-pointer -g -std=gnu99   -Werror=uninitialized -Werror=implicit-function-declaration -I. -I./ -I/usr/local/greenplum-db-6.26.3/include/postgresql/server -I/usr/local/greenplum-db-6.26.3/include/postgresql/internal -D_GNU_SOURCE -I/usr/include/libxml2   -c -o gpdb_6_9.4/src/backend/utils/adt/numutils.o gpdb_6_9.4/src/backend/utils/adt/numutils.c
gcc -m64 -Wall -Wmissing-prototypes -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fno-aggressive-loop-optimizations -Wno-unused-but-set-variable -Wno-address -m64 -O3 -fargument-noalias-global -fno-omit-frame-pointer -g -std=gnu99   -Werror=uninitialized -Werror=implicit-function-declaration -I. -I./ -I/usr/local/greenplum-db-6.26.3/include/postgresql/server -I/usr/local/greenplum-db-6.26.3/include/postgresql/internal -D_GNU_SOURCE -I/usr/include/libxml2   -c -o gpdb_6_9.4/src/backend/access/common/heaptuple.o gpdb_6_9.4/src/backend/access/common/heaptuple.c
gcc -m64 -Wall -Wmissing-prototypes -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fno-aggressive-loop-optimizations -Wno-unused-but-set-variable -Wno-address -m64 -O3 -fargument-noalias-global -fno-omit-frame-pointer -g -std=gnu99   -Werror=uninitialized -Werror=implicit-function-declaration -I. -I./ -I/usr/local/greenplum-db-6.26.3/include/postgresql/server -I/usr/local/greenplum-db-6.26.3/include/postgresql/internal -D_GNU_SOURCE -I/usr/include/libxml2   -c -o gpdb_6_9.4/src/backend/cdb/cdbappendonlystorageformat.o gpdb_6_9.4/src/backend/cdb/cdbappendonlystorageformat.c
gcc -m64 -Wall -Wmissing-prototypes -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fno-aggressive-loop-optimizations -Wno-unused-but-set-variable -Wno-address -m64 -O3 -fargument-noalias-global -fno-omit-frame-pointer -g -std=gnu99   -Werror=uninitialized -Werror=implicit-function-declaration -I. -I./ -I/usr/local/greenplum-db-6.26.3/include/postgresql/server -I/usr/local/greenplum-db-6.26.3/include/postgresql/internal -D_GNU_SOURCE -I/usr/include/libxml2   -c -o gpdb.o gpdb.c
gpdb.c: In function 'Dump_AppendOnlyStorage_ContentInfo':
gpdb.c:224:2: error: unknown type name 'Bytef'
  Bytef    *uncompressedDatum = NULL;
  ^
gpdb.c:326:13: warning: assignment from incompatible pointer type [enabled by default]
   headerPtr = uncompressedDatum;
             ^
make: *** [gpdb.o] Error 1
```

--- a/gpdb.c
+++ b/gpdb.c
@@ -222,7 +222,7 @@ Dump_AppendOnlyStorage_ContentInfo(uint8 *headerPtr,
 								   AppendOnlyStorageReadCurrent *blkHdr)
 {
 	DatumStreamBlock_Orig *contentPtr;
-	Bytef	   *uncompressedDatum = NULL;
+	uint8 	   *uncompressedDatum = NULL;
 	bool		success = true;
 
 	printf("\n\t<AO Content Info> -----");

